### ISSUE TITLE
Can't reproduce results reported in the paper

### ISSUE THREAD

**Issue by flackbash** [magic282/NQG#8]
I'm trying to reproduce the results reported in the paper but am getting a considerably lower BLEU score (by almost 1 BLEU point) on both the dev and test set.

I have run `run_squad_qg.sh` with the pre-defined parameters.
I used the generated model `model_e20.pt` and ran `translate.py` with the following parameters

```
python translate.py \
       -model "${MODELPATH}/model_e20.pt" \
       -src "${DATAPATH}/test/dev.txt.shuffle.test.source.txt" \
       -bio "${DATAPATH}/test/dev.txt.shuffle.test.bio" \
       -feats "${DATAPATH}/test/dev.txt.shuffle.test.pos" "${DATAPATH}/test/dev.txt.shuffle.test.ner" "${DATAPATH}/test/dev.txt.shuffle.test.case" \
       -tgt "${DATAPATH}/test/dev.txt.shuffle.test.target.txt" \
       -output "${SAVEPATH}/pred.txt" \
       -replace_unk \
       -verbose \
       -n_best 10 \
       -gpu 0
```

I then ran `test.py` in PyBLEU as 

```
python3 test.py ../../../../data/redistribute/QG/test/dev.txt.shuffle.test.target.txt ../../../../data/generated/pred.txt
```
which resulted in a BLEU score of 11.26 which is almost one bleu score lower than the reported NQG+ BLEU score for the test set in the paper (12.18).

Do you have an explanation or an intuition for this considerable discrepancy? Maybe the parameters in `run_squad_qg.sh` are not those that were used to compute the results for the paper?

And thanks a lot for making this code public :)

**Comment by magic282** [comment]
Hi, I guess it would be better to first pick a model based on the performance on the dev set and then do the testing. And I guess the model checkpoint on epoch 20 is already overfitted. Thank you.

**Comment by flackbash** [comment]
Thanks for the lead!
In a new experiment, I took the best performing model on the dev set (`model_e6.pt`) and ran `translate.py` over it.
Additionally, I removed the copy marks ("[[" and "]]") in the predicted questions in `test.py`
This now gives me a BLEU score of 13.79 on the dev set and 13.71 on the test set which is better even than the reported BLEU scores for the best performing architecture NQG++.
Do you have any additional explanations for the discrepancy?

**Comment by magic282** [comment]
Hi, 13.71 is expected for this PyTorch code. The paper reports an very old result which was implemented using one Microsoft in-house deeplearning tool.

**Comment by flackbash** [comment]
Ok perfect, thanks :)
